services/blockchain-indexer/shared/utils/eventsUtils.js

```diff
@@ -52,6 +74,23 @@ const getEventsInfoToIndex = async (block, events) => {
 	return eventsInfoToIndex;
 };
 
+const deleteEventsTillBlockHeight = async (blockHeight, dbTrx) => {
+	if (Number.isNaN(Number(blockHeight))) return;
+
+	const eventsTable = await getEventsTable();
+	const eventTopicsTable = await getEventTopicsTable();
+	const queryParams = {
+		propBetweens: [{
+			property: 'height',
+			to: blockHeight,
+		}],
+		limit: 10000,
```

Comment:
Why the `limit`? What if there were more than 10000 eligible entries?
```suggestion
```

Reply:
those will be deleted eventually on calls for each next block. the reason to add limit is the following case - suppose blocks were indexed with `db.config.db.isPersistEvents = true` before and now the user set `config.db.isPersistEvents = false` and restarted indexer. Now this call will try to delete all the events and event topics of finalized blocks stored so far which can be a expensive operation. this limit will ease the load by cleaning up in chunks for each subsequent blocks.